File: src/test/java/com/mkreidl/ephemeris/solarsystem/EquationOfTimeTest.java
package com.mkreidl.ephemeris.solarsystem;

import com.mkreidl.ephemeris.Time;
import com.mkreidl.ephemeris.geometry.Angle;

import org.junit.Assert;
import org.junit.Test;

import java.util.Calendar;
import java.util.GregorianCalendar;
import java.util.TimeZone;

public class EquationOfTimeTest
{
    @Test
    public void testEquationOfTimeMeeusChap28()
    {
        final Calendar calendar = new GregorianCalendar( TimeZone.getTimeZone( "UTC" ) );
        calendar.set( 1992, 9, 13, 0, 0, 0 );
        final Time time = new Time( calendar.getTimeInMillis() );
        final double radians = Angle.standardize( Sun.getEquationOfTime( time ) );
        Assert.assertEquals( 13 + 42.6 / 60, Math.toDegrees( radians ) * 4, 0.02 );
    }
}

File: src/test/java/com/mkreidl/ephemeris/solarsystem/EquationOfTimeTest.kt
package com.mkreidl.ephemeris.solarsystem

import com.mkreidl.ephemeris.Instant
import org.junit.Assert
import org.junit.Test

class EquationOfTimeTest {
    private val october13_1992 = Instant.ofJulianDayFraction(2_448_908.50)  // October 13, 1992, 0hTD

    @Test
    internal fun testApparentLongitude() {
        Assert.assertEquals(Math.toRadians(-161.619_18), SunLowPrecision(october13_1992).apparentLongitude, 1e-6)
    }

    @Test
    internal fun testEquationOfTimeMeeusChap28() {
        val radians = SunLowPrecision(october13_1992).equationOfTime
        Assert.assertEquals(13 + 42.6 / 60, Math.toDegrees(radians) * 4, 0.002)
    }
}
